### PR TITLE
Fix Proc#curry signature

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: core/proc.rbs [hunk 1]
```diff
@@ -262,7 +262,7 @@ class Proc < Object
 
   def []: (*untyped arg0) -> untyped
 
-  def curry: (?Integer arity) -> Proc
+  def curry: (?_ToInt arity) -> Proc
```
Comment (kachick) [inline core/proc.rbs:265]:
```suggestion
  def curry: (?int arity) -> Proc
```

`int` looks work too, both is same? Then which one should be preferred? 🤔

Comment (soutaro) [reply core/proc.rbs:265]: It's equivalent because `_ToInt` is `int`.

Comment (kachick) [reply core/proc.rbs:265]: Sure!

**Comment by kachick** [comment]
Thank you!

**Pull request merged**